### PR TITLE
DOC: Installation instructions update

### PR DESCRIPTION
**Pull request by VISWESWARAN1998**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by VISWESWARAN1998** [comment]
I followed all the steps specified in the ReadMe documentation. I have installed MiniForge, Mamba and checkout the repository. 

I then followed the installation instructions specified in the README.md file and executed the below command

```sh
python -m pip install -ve . --no-build-isolation --config-settings=editable-verbose=true
```
And was facing several errors. After inspecting the logs I found out that they were ModuleNotFound errors so I inspected more and found I have to install all the requirements first

```sh
python -m pip install -r requirements-dev.txt
```

I have updated the ReadMe accordingly so, new comers won't stumble like me!

Updated document will have the following commands now

```sh
python -m pip install -r requirements-dev.txt
python -m pip install -ve . --no-build-isolation --config-settings=editable-verbose=true
```

**Comment by datapythonista** [comment]
This is incorrect, sorry. The instructions are for general users, you are making them install the dependencies for people who contributes to pandas, which doesn't make sense. Closing.

**Comment by VISWESWARAN1998** [comment]
It had this instructions for development mode which I tried and it failed. Instructions were partial

or for installing in [development mode](https://pip.pypa.io/en/latest/cli/pip_install/#install-editable):

python -m pip install -ve . --no-build-isolation --config-settings=editable-verbose=true

**Comment by datapythonista** [comment]
You are correct, sorry, I reviewed this PR too quick.

@lithomas1, I think this is failing since we moved to Meson. Shouldn't pip be able to find the dependencies it needs? When installing in editable mode pip fails because `mesonpy` is not found, and after installing it manually it fails because ofther deps like jinja.

File: README.md [hunk 1]
```diff
@@ -137,6 +137,7 @@ or for installing in [development mode](https://pip.pypa.io/en/latest/cli/pip_in
 
 
 ```sh
+python -m pip install -r requirements-dev.txt
 python -m pip install -ve . --no-build-isolation --config-settings=editable-verbose=true
```
Comment (lithomas1) [inline README.md:141]: If you don't need an editable installation, you can leave out the ``--no-build-isolation`` part
(which will then let pip pull dependencies for you)

I'm thinking that we should really remove the whole development mode section from the README, though.
You probably don't want to do this unless you want to work on pandas itself, and I think the docs here are much better at explaining things.
https://pandas.pydata.org/docs/development/contributing_codebase.html

**Comment by VISWESWARAN1998** [comment]
Should I remove and send pull request?

**Comment by datapythonista** [comment]
Yes, feel free to edit this same branch/PR and remove that block.

File: README.md [hunk 2]
```diff
@@ -133,12 +133,7 @@ cloning the git repo), execute:
 pip install .
 ```
 
-or for installing in [development mode](https://pip.pypa.io/en/latest/cli/pip_install/#install-editable):
-
-
-```sh
-python -m pip install -ve . --no-build-isolation --config-settings=editable-verbose=true
-```
+or for installing in [development mode](https://pip.pypa.io/en/latest/cli/pip_install/#install-editable).
```
Comment (lithomas1) [inline README.md:136]: Can you delete this block?

The line below telling users how to install from source should be enough.

**Pull request closed without merging**